synapse/config/_base.py

```diff
@@ -876,4 +876,20 @@ def get_instance(self, key: str) -> str:
         return self.instances[remainder]
 
 
+@attr.s
+class RoutableShardedWorkerHandlingConfig(ShardedWorkerHandlingConfig):
+    """A version of `ShardedWorkerHandlingConfig` that is used for config
+    options where all instances know which instances are responsible for the
+    sharded work.
+    """
+
+    def __attrs_post_init__(self):
+        # We require that `self.instances` is non-empty.
+        assert self.instances
```

Comment:
Should these raise `ConfigError` instead? (Or at least provide a message for how the user can fix this?)

Reply:
We should be checking before this, as we can't give a particularly helpful message. I'll make it raise a slightly more useful exception though.

Relatedly, looks like we don't actually check that there is at least one event stream writer, I'll fix that too.